examples/sdk-interactive-examples/scripts/createExample.mjs

```diff
@@ -0,0 +1,120 @@
+
+    // Must be a valid directory name
+    // For simplicity, we assume a valid directory name doesn't contain the following characters: \ / : * ? " < > |
+    const isValidChars = !/[\\/:*?"<>|]/.test(directoryName);
+
+    return noWhiteSpaces && startsWithExample && isValidChars;
+}
+rl.question('Enter example directory name (should have prefix "example-"): ', (name) => {
+    if (fs.existsSync(path.join(EXAMPLES_DIR, name))) {
+        console.error('Error: This example directory already exists.');
+        process.exit(1);
+    }
+
+    if(!isValidDirectoryName(name)){
+        console.error('Error: Must not contain white spaces or spaces.');
+        console.error('Error: Must start with "example-"');
+        console.error('Error: Must be a valid directory name does not contain the following characters: \ / : * ? " < > |');
+        process.exit(1);
+    }
+
+    exampleName = name;
+
+    rl.question('Enter example title: ', (title) => {
+        exampleTitle = title;
+
+        rl.question('Enter example description: ', (description) => {
+            exampleDescription = description;
+
+            const NEW_EXAMPLE_DIR = path.join(EXAMPLES_DIR, exampleName);
+
+            // Create new example directory and copy template
+            execSync(`rsync -av --progress ${TEMPLATE_DIR}/ ${NEW_EXAMPLE_DIR}/`);
+
+            // Update package.json
+            const packageJsonPath = path.join(NEW_EXAMPLE_DIR, 'package.json');
+            const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
+            packageJson.name = EXAMPLE_PACKAGE_NAME_PREFIX+exampleName;
+            packageJson.description = exampleDescription;
+            packageJson.title = exampleTitle;
+            fs.writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 2)+'\n');
+            
+
+            // Update ./.codesandbox/template.json
+            const templateJsonPath = path.join(NEW_EXAMPLE_DIR, '.codesandbox', 'template.json');
+            const templateJson = JSON.parse(fs.readFileSync(templateJsonPath, 'utf-8'));
+            templateJson.title = exampleTitle;
+            templateJson.description = exampleDescription;
+
+            fs.writeFileSync(templateJsonPath, JSON.stringify(templateJson, null, 2));
+           
+            //Update ./README.md
+
+            const EXAMPLE_README_PATH = path.join(NEW_EXAMPLE_DIR, 'README.md');
+            let readme = fs.readFileSync(EXAMPLE_README_PATH, 'utf-8');
+            readme = readme.replaceAll('{EXAMPLE-TITLE}', exampleTitle);
+            readme = readme.replaceAll('{EXAMPLE-DESCRIPTION}', exampleDescription);
+        
+            const exampleSandBoxPath = EXAMPLE_CODESANDBOX_PATH_TEMPLATE.replace('{EXAMPLE-DIR}', exampleName);
+
+            readme = readme.replaceAll('{CODESANDBOX-PATH}', exampleSandBoxPath);
+        
+            fs.writeFileSync(EXAMPLE_README_PATH, readme);
+
+            // This is ugly
```

Comment:
I would delete this comment. Either make it nicer, or just delete it :)

Reply:
thx